### PR TITLE
build: Upgrade hatchling to 1.20.0

### PR DESCRIPTION
**Pull request by lukpueh**

### CODE REVIEW

**Comment by lukpueh** [comment]
IMO, we can wait for a couple of days before we merge in case there comes another patch release.

**Review by jku — APPROVED** [review]
quick manual test says this works, thanks (and the hatchling update seems fine too). 

It's annoying that `build`  does not document the isolated build environment versions in the log but it looks like it does the right thing.

**Comment by lukpueh** [comment]
> It's annoying that build does not document the isolated build environment versions in the log but it looks like it does the right thing.

Yep, and there is no sane way of making it more verbose.  I did check if it works, by removing the stdout pipes in the [`subprocess` call used to run `pip install`](https://github.com/pypa/build/blob/de5b44b0c28c598524832dff685a98d5a5148c44/src/build/env.py#L64). 🙈

**Pull request merged**
